README.md

```diff
@@ -36,13 +36,108 @@ Examples:
   validate-ddf ../ddf-example --exclude-tags "WARNING_TAG"           Get all kinds of issues except warnings
 ```
 
-### index file creation
+## API usage
+
+First of all you should install this package: `npm i ddf-validation`
+
+API for `ddf-validation` has one method that expects particular type of validator and 3 types of validator:
+
+ * JSON based validator (`JSONValidator`)
+ * stream based validator (`StreamValidator`)
+ * validator that returns only kind of current DDF dataset: `correct/true` or `incorrect/false`
+ 
+Some examples of API using:
+
+### JSONValidator
+
+Simple example
+
+```
+const api = require('ddf-validation');
+const jsonValidator = new JSONValidator('path to ddf dataset');
+
+jsonValidator.on('finish', (err, jsonIssuesContent) => {
+  console.log(err, jsonIssuesContent);
+});
+
+api.validate(jsonValidator);
+```
+
+This validator's type returns all issues as JSON object. 
+And for this reason it not will be good for huge DDF datasets.
+
+### StreamValidator
+
+```
+const api = require('ddf-validation');
+const streamValidator = new StreamValidator('path to ddf dataset');
+
+streamValidator.on('issue', () => {
+  // catch new issue here
+});
+
+streamValidator.on('finish', err => {
+  // validation is finished
+});
+
+api.validate(streamValidator);
+```
+
+This validator's type returns each issue separately one by one.
+And for this reason it will be good for huge DDF datasets.
+By the way `StreamValidator` is used in console utility (`validate-ddf`) of this package.
+
+### FundamentalValidator
+
+This validator's type returns only `true` or `false` in accordance with kind of DDF dataset.
```

Comment:
According to the state of the dataset (valid or not) this validator returns only `true` or `false` with appropriate meaning